### DISCUSSION TITLE
Restrict User visibility

### DISCUSSION THREAD

**Discussion by Atreb92** [photoprism/photoprism#3729] [Q&A] [answered]
Hi,
I just enabled PhotoPrism Essentials to try out user roles: is there any intended way for restricting a user visibility cone to only a subset of original folders? I tried looking into the documentation and settings and could only find something similar in the Pro section.
Thank you for the information!

**Comment by graciousgrey** [comment, marked as answer]
Guest users can only see content that has previously been shared with them. 

See: 
- https://docs.photoprism.app/user-guide/users/roles/#guest
- https://docs.photoprism.app/user-guide/users/sharing/

At the moment you need to use sharing links to share content with guest users. In a future release, you will be able to share content with local users directly from the web interface without having to [create links](https://docs.photoprism.app/user-guide/share/) first.

You might also be interested in:
- https://docs.photoprism.app/user-guide/users/libraries/

**Reply by Atreb92** [reply]
Thanks for the reply. 
Unfortunatly, I don't think the first solution of guest would fit my use case. I currently host my whole family photos on my server; which include both my personally taken ones and old archived ones I have collected around. My SO parents would like to have access to the photos I host for them but I cannot find an easy way to do this other than create a second PhotoPrism container/instance with only that library in scope (which would require more resources and storage, like thumbnails, for essentially the same photos). 
Maybe multi-user libraries you linked could work in this case but as far as I understand its a "Plus" only feature. I might have missed it, but the comparison on the website does not include this feature on the difference table so I personally was not aware (or missed it) when deciding which tier to go with.

**Reply by graciousgrey** [reply]
> I might have missed it, but the comparison on the website does not include this feature on the difference table so I personally was not aware (or missed it) when deciding which tier to go with.

The library manager is not implemented yet, that's why it does not appear on the feature comparison. At this time you would need to run multiple instances if you want users with separated libraries. 

However you could create albums from the folders your family should get access to and share them with their guest account.

**Reply by Atreb92** [reply]
Oh, makes sense.
Anyway, I tried with creating an album with all photos from the subfolder i want them to access. It is mostly OK but at the same time kind of limited since the guest has very limited features. I will probably go with the second instance at this point until future features get added. Thank you very much for your support!

**Reply by graciousgrey** [reply]
You are welcome :)
